docs/rfcs/0009-remove-interactors-and-controllers.md

```diff
@@ -0,0 +1,375 @@
+            Pocket.homeRecsCategoryClicked.record(
+                Pocket.HomeRecsCategoryClickedExtra(
+                    categoryName = action.categoryName,
+                    newState = "deselected",
+                    selectedTotal = currentCategoriesSelections.size.toString(),
+                ),
+            )
+        } else {
+            Pocket.homeRecsCategoryClicked.record(
+                Pocket.HomeRecsCategoryClickedExtra(
+                    categoryName = action.categoryName,
+                    newState = "selected",
+                    selectedTotal = currentCategoriesSelections.size.toString(),
+                ),
+            )
+        }
+    }
+}
+
+
+// finally, we can shift all the state updating logic into the reducer
+fun reduce(state: AppState, action: AppAction): AppState = when (action) {
+    ...
+    is AppAction.TogglePocketStoriesCategory -> {
+        val currentCategoriesSelections = state.pocketStoriesCategoriesSelections.map { it.name }
+        val updatedCategoriesState = when {
+            currentCategoriesSelections.contains(action.categoryName) -> {
+                state.copy(
+                    pocketStoriesCategoriesSelections = state.pocketStoriesCategoriesSelections.filterNot {
+                        it.name == action.categoryName
+                    },
+                )
+            }
+            currentCategoriesSelections.size == POCKET_CATEGORIES_SELECTED_AT_A_TIME_COUNT -> {
+                state.copy(
+                    pocketStoriesCategoriesSelections = state.pocketStoriesCategoriesSelections
+                        .minusOldest()
+                        .plus(
+                            PocketRecommendedStoriesSelectedCategory(
+                                name = action.categoryName,
+                            )
+                        ),
+                )
+            }
+            else -> {
+                state.copy(
+                    pocketStoriesCategoriesSelections =
+                    state.pocketStoriesCategoriesSelections +
+                            PocketRecommendedStoriesSelectedCategory(action.categoryName)
+                )
+            }
+        }
+        // Selecting a category means the stories to be displayed needs to also be changed.
+        updatedCategoriesState.copy(
+            pocketStories = updatedCategoriesState.getFilteredStories(),
+        )
+    }
+}
+```
+
+---
+
+### Interacting with the storage layer through middlewares
+
+What if a feature requires some async data for its initial state, or needs to write changes to disk? There are no restrictions on running impure methods in middleware, so we can respond to an actions that would changes to the storage layer from middlewares. A good example is [ContainerMiddleware](../../android-components/components/feature/containers/src/main/java/mozilla/components/feature/containers/ContainerMiddleware.kt)
```

Comment:
Ah, good point. I had thought to use a relative link for the exact opposite reason - so that it would reference the up-to-date version. I suppose that doesn't account for the path breaking either though